src/reference/90-Developers-Guide/02-Coding-Guideline/00.md

```diff
@@ -90,83 +90,87 @@ Use [MiMa][mima].
 
 #### Public traits should contain `def` declarations only
 
+TODO: "results to codegen at subclass" is unclear
```

Comment:
The intent is that vals/vars/lazy vals are: (1) problematic for trait lineralisation, (2) problematic for binary compatibility.

But I'm not sure what exactly are and how to exactly explain the specific technical reasons behind it.